src/modules/csm/state.py

```diff
@@ -169,7 +164,7 @@ def migrate(
         else:
             self.data = {frame: defaultdict(AttestationsAccumulator) for frame in frames}
 
-        self._epochs_per_frame = epochs_per_frame
+        self.frames = frames
         self._epochs_to_process = tuple(sequence(l_epoch, r_epoch))
         self._consensus_version = consensus_version
         self.find_frame.cache_clear()
```

Comment:
Let move it closer to line 167. Setting new frames, then clearing the cache. Or vice versa.